api/resource/resource.go

```diff
@@ -26,7 +25,7 @@ import (
 // paired with metadata used by kustomize.
 // For more history, see sigs.k8s.io/kustomize/api/ifc.Unstructured
 type Resource struct {
```

Comment:
`Resource.AsYAML` should be changed to directly return YAML instead of marshal to json at first.

Reply:
Yes, we should try that
Filed #3722
That's a huge PR in terms of tests edits, because fields will be moved around, so should probably do it separately.
Hope you want to take it on
We should get some benchmarks in place, which say do the krusty/mediumoverlay_test (or something better) 100 times.